controllers/argocd/applicationset.go

```diff
@@ -764,9 +764,12 @@ func getApplicationSetContainerImage(cr *argoproj.ArgoCD) string {
 	tag := ""
 
 	// First pull from spec, if it exists
-	if cr.Spec.ApplicationSet != nil {
+	if cr.Spec.ApplicationSet.Image != "" && cr.Spec.ApplicationSet.Version != "" {
 		img = cr.Spec.ApplicationSet.Image
 		tag = cr.Spec.ApplicationSet.Version
+	} else if cr.Spec.Image != "" && cr.Spec.Version != "" && cr.Spec.ApplicationSet.Image == "" && cr.Spec.ApplicationSet.Version == "" {
+		img = cr.Spec.Image
+		tag = cr.Spec.Version
```

Comment:
Can this be simplified? Refer https://github.com/argoproj-labs/argocd-operator/blob/f8b8fab31f9df9f1a47f40e11b1b09879fe19e21/controllers/argocd/util.go#L222-L240

Reply:
From the above code it seems it can take image and tag from two different places, e.g. if only `cr.Spec.Repo.Image`  field is defined, it will take the image from the field but use the __default__ version if `cr.Spec.Repo.Version`  or `cr.Spec.Version` is not defined. 

I think that should not be allowed, as it might just fail.

One thing that I think is missing in my code is logging, I think it should log the error that image is not used if the version was missing. but that might just be over engineering.

Reply:
Yeah. Let's assume the user will handle cases where it fails due to an incorrect image. Usually, the deployment will fail with an appropriate `ImagePullBackOff` error to indicate this. We should also consider cases where the user is simply interested in trying out a different version of the image, so not having a strict requirement between the image and version fields would be helpful. WDYT?

Reply:
Yeah, make sense...